### ISSUE TITLE
Pull in changes from keyBld repository

### ISSUE THREAD

**Issue by ghost** [KeyWorksRW/ttLib#87]
### Description:
This issue category should be called "This was a great idea until it wasn't".

At one point, it looked like the **MakeNina** project was nearing completion. I suspected that it wouldn't see much use outside of KeyWorks projects, and since all KeyWorks projects require **ttLib** as a sub project, it seemed to make sense to merge in **MakeNinja** as an optional KeyWorks tool. I then created a merged repository and continued development there, archiving this repository. That worked fine for a brief while until I decided to make numerous significant changes to **MakeNinja** both to add extensive support for **VS Code**, but also with overhaul of how custom builds are setup, plans for supporting the GCC compiler, plans to make it fully functional as a cross-platform tool, etc. That means lots and lots of issues being raised and fixes having nothing to do with **ttLib**, and that looks like it will continue for some time. In addition, **MakeNinja**'s support for **VS Code**, particularly when run on Windows should make it quite useful for any kind of C/C++ repository, not just the KeyWorks ones. That means once it becomes public, it could potentially see a lot of changes from other devs besides the KeyWorks teams, and that would result in frequent updates to the **ttLib** sub module, even though **ttLib** itself won't change a lot.

Bottom line is, I'm reactivating this repository and copying over the changes to ttLib. This issue is to cover that fact that several changes are being made without specific issue numbers (they will be in the old merged repository).

### Issue Type:
- [ ] Bug report
- [ ] Feature request
- [x] Other
